corehq/apps/case_importer/do_import.py

```diff
@@ -528,12 +528,15 @@ def _get_external_id(self):
         return self.external_id
 
     def _get_caseblock_kwargs(self):
-        return {
+        kwargs = {
             'update': self.fields_to_update,
             'index': self._get_parent_index(),
-            'date_opened': self._get_date_opened() or CaseBlock.undefined,
-            'external_id': self._get_external_id() or CaseBlock.undefined,
         }
+        if date_opened := self._get_date_opened():
+            kwargs['date_opened'] = date_opened
+        if external_id := self._get_external_id():
+            kwargs['external_id'] = external_id
+        return kwargs
```

Comment:
totally random thought - we use this code pattern a lot, and something about it always irks me for the loss of a declarative feel.  In javascript these two things are essentially equivalent:
```javascript
var options = {},
    date_opened = get_date_opened();
if (date_opened) {
    options.date_opened = date_opened
}

// is about the same as
var options = {
    date_opened: get_date_opened() || undefined,
};
```
Whereas in Python we're stuck with the former option.  An easy, but not very clear alternative is
```python
options = dict((k, v) for k, v in {
    'date_opened': self._get_date_opened()
}.items() if v)
```
Of course that assumes all falsy values should be removed, so perhaps a safer bet would be to let each row determine when it should be included, like this:
```python
options = dict((k, v) for k, v in {
    'date_opened': self._get_date_opened() or ...
}.items() if v is ...)
```
I wonder if this would be good to encapsulate in some way, so we could instead do

```python
options = strip_ellipsis({
    'date_opened': self._get_date_opened() or ...
})
```
Perhaps with an optional second arg determining whether to strip `...`, `None`, or all falsy values.

It's not directly related to this PR, but it feels adjacent enough to pose the question.